10/umbraco-deploy/deployment-workflow/import-export.md

```diff
@@ -0,0 +1,440 @@
+        {
+            if (value is not null)
+            {
+                _logger.LogWarning("Skipping migration of Nested Content items ({PropertyTypeAlias}), because value could not be parsed: {Value}.", propertyType.Alias, value);
+            }
+
+            return null;
+        }
+
+        var layoutItems = new List<BlockListLayoutItem>();
+        var contentData = new List<BlockItemData>();
+
+        foreach (NestedContentItem nestedContentItem in nestedContentItems)
+        {
+            IContentType? contentType = contextCache.GetContentTypeByAlias(_contentTypeService, nestedContentItem.ContentTypeAlias);
+            if (contentType is null)
+            {
+                _logger.LogWarning("Skipping migration of Nested Content item ({Id}), because content type does not exist: {ContentTypeAlias}.", nestedContentItem.Id, nestedContentItem.ContentTypeAlias);
+                continue;
+            }
+
+            var udi = new GuidUdi(Constants.UdiEntityType.Element, nestedContentItem.Id);
+
+            layoutItems.Add(new BlockListLayoutItem()
+            {
+                ContentUdi = udi
+            });
+
+            contentData.Add(new BlockItemData()
+            {
+                Udi = udi,
+                ContentTypeKey = contentType.Key,
+                RawPropertyValues = nestedContentItem.RawPropertyValues
+            });
+        }
+
+        var blockValue = new BlockValue()
+        {
+            Layout = new Dictionary<string, JToken>()
+            {
+                { Constants.PropertyEditors.Aliases.BlockList, JToken.FromObject(layoutItems) }
+            },
+            ContentData = contentData
+        };
+
+        return JsonConvert.SerializeObject(blockValue, Formatting.None);
+    }
+}
+```
+
+Moving forward, other migrators may be built by HQ or the community for property editors found in community packages. We'll make them available for [use](https://www.nuget.org/packages/Umbraco.Deploy.Contrib) and [review](https://github.com/umbraco/Umbraco.Deploy.Contrib) via the `Umbraco.Deploy.Contrib` package.
+
+### Migrating from Umbraco 7
+
+The import and export feature is available from Deploy 4 (which supports Umbraco 8). It's not been ported back to Umbraco 7, hence you can't trigger an export from there in the same way.
+
+We are still able to use this feature though to help migration from Umbraco 7 to a more recent major version.
+
+We can generate an export zip file in the same format as that used by the content import/export feature. With that we can import it into Umbraco 8 or above.  And apply similar migrations to update obsolete data types and property data into newer equivalents.
+
+This is possible via code - by temporarily applying a composer to an Umbraco 7 solution to generate the export file on start-up.
+
+An example follows:
+
+[TBC - code sample]
```

Comment:
I still have to create a 'good enough' example and verify whether that can be successfully imported in newer versions as well. Maybe we should add this section at a later date?

Reply:
OK.  Are you happy if I leave this as "An example will be available soon."?  Just as I would like to at least include in the blog and docs that we have thought about this and plan to include.

Reply:
I've done that for the blog post and docs - we can add the sample to the latter next week when it's ready.